packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -645,6 +649,93 @@ describe("Garbage Collection Tests", () => {
 				);
 			});
 
+			/**
+			 * Here, the base snapshot contains nodes that have timed out. The test validates that we generate errors
+			 * when these nodes are used via trailing ops. It simulates trailing ops by ensuring that garbage collector
+			 * is not loaded and GC does not run before nodeUsed is called.
+			 */
+			it("generates events for nodes that time out on load - nodes are used via trailing ops", async () => {
```

Comment:
This test is a little hard to follow - it feels better suited as an e2e test where there's less in-depth manipulation of stuff.  I see several comments about simulating or doing something explicitly that other code will do in a typical setup, which feels potentially brittle.

Reply:
I think this scenario is covered by the trailing ops e2e tests, so I assume your point is to get Unit Test coverage.  Maybe you can explain a little more about the benefits of this approach.

Reply:
Yes, these scenarios are hard to simulate both here and in end-to-end tests. As unit tests are simpler to write, I chose to do it here.

> I see several comments about simulating or doing something explicitly that other code will do in a typical setup, which feels potentially brittle.

Isn't that what unit tests do? I agree that some of the simulation are not very apparent and we should definitely clean those up.